internal/pkg/cli/file/watch.go

```diff
@@ -0,0 +1,106 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package file
+
+import (
+	"io/fs"
+	"path/filepath"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+// RecursiveWatcher wraps an fsnotify Watcher to recursively watch all files in a directory.
+type RecursiveWatcher struct {
+	fsnotifyWatcher *fsnotify.Watcher
+	done            chan struct{}
+	closed          bool
+	events          chan fsnotify.Event
+	errors          chan error
+}
+
+// NewRecursiveWatcher returns a RecursiveWatcher which notifies when changes are made to files inside a recursive directory tree.
+func NewRecursiveWatcher(buffer uint) (*RecursiveWatcher, error) {
+	watcher, err := fsnotify.NewBufferedWatcher(buffer)
+	if err != nil {
+		return nil, err
+	}
+
+	rw := &RecursiveWatcher{
+		events:          make(chan fsnotify.Event, buffer),
+		errors:          make(chan error),
+		fsnotifyWatcher: watcher,
+		done:            make(chan struct{}),
+		closed:          false,
+	}
+
+	go rw.start()
+
+	return rw, nil
+}
+
+// Add recursively adds a directory tree to the list of watched files.
+func (rw *RecursiveWatcher) Add(path string) error {
+	if rw.closed {
+		return fsnotify.ErrClosed
+	}
+	return filepath.WalkDir(path, func(p string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return nil
+		}
```

Comment:
Why do we ignore the error 🤔 
```suggestion
		if err != nil {
			return err
		}
```

Reply:
Because this is an error from walking the directory, in which case we just want to ignore the directory/file it's looking at. Adding a comment to make this more clear.